### PR TITLE
Improve error messages for Docker container-image builds

### LINKED ISSUE #8523: container-image-docker silently fails
**Describe the bug**
If Docker daemon is not available, build finishes with success even if Docker image creation fails. This is big issue in CI environment.

```
 [INFO] --- quarkus-maven-plugin:1.3.2.Final:build (default) @ asc-api-service ---
 [INFO] Init
 [WARNING] No etcd endpoint configured. Source will be disabled ;_;
 [INFO] Initialization finished
 [WARNING] [io.quarkus.container.image.docker.deployment.DockerBuild] Could not connect to docker daemon!
 [WARNING] [io.quarkus.container.image.docker.deployment.DockerBuild] Could not connect to docker daemon!
 [INFO] [org.jboss.threads] JBoss Threads version 3.0.1.Final
 [INFO] [io.quarkus.smallrye-reactive-messaging.deployment.processor] Metric is disabled - vetoing the MetricDecorator
 [INFO] Initializing dekorate session.
 [INFO] [org.hibernate.Version] HHH000412: Hibernate ORM core version 5.4.14.Final
 [INFO] Default s2i build generator....
 [INFO] Registering s2i handler!
 [INFO] Registering s2i handler!
 [INFO] Generating manifests.
 [INFO] Processing kubernetes configuration.
 [INFO] Processing openshift configuration.
 [INFO] Processing s2i configuration.
 [INFO] Closing dekorate session.
 [INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: /builds/myproject/mymodule/target/service-runner.jar
 [INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 3899ms
 [INFO] ------------------------------------------------------------------------
 [INFO] BUILD SUCCESS
 [INFO] ------------------------------------------------------------------------
```
No Docker image was generated.

**Expected behavior**
(Describe the expected behavior clearly and concisely.)
Build should fail if image cannot be created.

**Actual behavior**
(Describe the actual behavior clearly and concisely.)
Build finishes successfully, no image was created.

**To Reproduce**
1. Kill docker daemon
2. mvn package

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.3.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #8523

### CODE REVIEW

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerWorking.java [hunk 1]
```diff
@@ -11,38 +11,35 @@
...
-    }
+    private static final Logger LOGGER = Logger.getLogger(DockerWorking.class.getName());
 
     @Override
     public boolean getAsBoolean() {
-        // No need to perform the check multiple times.
-        if (daemonFound) {
-            return true;
+        try {
+            if (!ExecUtil.execSilent("docker", "help")) {
```
Comment (gastaldi) [inline extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerWorking.java:23]: This returns a single line and can be faster than reading the whole output of `docker help`
```suggestion
            if (!ExecUtil.execSilent("docker", "-v")) {
```

Comment (geoand) [reply extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerWorking.java:23]: Good suggestion! I was looking for something like that and I totally missed it!

Comment (geoand) [reply extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerWorking.java]: Fixed

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**